### PR TITLE
NERSC Programming Environment prototype

### PR DESCRIPTION
**Pull request by kngott**
## Summary

Adds NERSC's programming environment prototype on Perlmutter to the make system.

NPE is experimental at this time.

## Additional background

I'm using it to compile with a more up-to-date MPICH version. Makes sense to add it to the repo for future analyses and others to use, if interested.

To use:
```
module use /global/cfs/cdirs/nstaff/cookbg/pe/modulefiles
module load npe
module load mpich
module unload cray-libsci darshan
```

Tested compiling in the GNU environment with CUDA.

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
@kngott Could you fix the conflicts?

**Comment by kngott** [comment]
Done. Surprising that triggered anything at all.

File: Tools/GNUMake/sites/Make.nersc [hunk 1]
```diff
@@ -25,11 +25,16 @@ ifeq ($(which_computer),$(filter $(which_computer),perlmutter))
   endif
 
   ifeq ($(USE_CUDA),TRUE)
+    ifdef NPE_VERSION
+      CFLAGS += -Xcompiler="$(filter-out -Wl%, $(wordlist 2,1024,$(shell mpicc -show 2> /dev/null))))"
```
Comment (WeiqunZhang) [inline Tools/GNUMake/sites/Make.nersc]: I think there is an extra `)` here.

Comment (kngott) [reply Tools/GNUMake/sites/Make.nersc]: Good catch! Fixed.

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**